### ISSUE TITLE
`update_lr` divides by zero when `lr_warmup_steps == 0` and `total_steps == 0`

### ISSUE THREAD

**Issue by danielbennett888** [sapientinc/HRM-Text#23]
In `pretrain.py:191-203`:

```python
def update_lr(config: PretrainConfig, train_state: TrainState) -> float:
    if train_state.step < config.lr_warmup_steps:
        lr = config.lr * min(1.0, train_state.step / config.lr_warmup_steps)
    else:
        progress = (train_state.step - config.lr_warmup_steps) / (train_state.total_steps - config.lr_warmup_steps)
        ...
```

When `lr_warmup_steps == 0` (which is the default in `config/cfg_sft.yaml`), the first branch is skipped at step 0, so the `else` branch runs with `progress = 0 / total_steps`, fine.

However the more interesting edge case is `total_steps == config.lr_warmup_steps`. With short SFT runs (e.g. tiny dataset) `total_steps` is computed at `pretrain.py:179` as:

```python
total_steps = int(config.epochs * train_metadata.total_length // config.global_batch_size)
```

If your fine-tune dataset is small enough that `total_steps <= lr_warmup_steps`, you get `progress = N / 0` (ZeroDivisionError) the moment the loop steps past warmup. This is unlikely on production runs but very easy to hit on smoke tests and short SFT trials.

A guard like `denom = max(1, train_state.total_steps - config.lr_warmup_steps)` would make this robust.
